src/sinks/loki.rs

```diff
@@ -786,6 +788,33 @@ mod integration_tests {
         .await;
     }
 
+    #[tokio::test]
+    async fn out_of_order_per_partition() {
+        let batch_size = 2;
+        let big_lines = random_lines(1_000_000).take(2);
+        let small_lines = random_lines(1).take(20);
+        let mut events = big_lines
+            .into_iter()
+            .chain(small_lines)
+            .map(Event::from)
+            .collect::<Vec<_>>();
+
+        let base = chrono::Utc::now() - Duration::seconds(30);
+        for (i, event) in events.iter_mut().enumerate() {
+            let log = event.as_mut_log();
+            log.insert(
+                log_schema().timestamp_key(),
+                base + Duration::seconds(i as i64),
+            );
+        }
+
+        // So, since all of the events are of the same partition, and if there is concurrency,
+        // then if ordering inside paritions isn't upheld, the big line events will take longer
+        // time to flush than small line events so loki will receive smaller ones before large
+        // ones hence out of order events.
+        test_out_of_order_events(OutOfOrderAction::Drop, batch_size, events.clone(), events).await;
```

Comment:
It is. 

>  Is there any way to provoke out of order events without racing

Not really. There are two more ways to provoke them:
* Fail requests and force them to retry. 
   * This isn't quite testing what it needs too since retry logic can correct this.
   * A mock loki server which we can configure to fail requests would be needed.
* Stall the start of the server so that requests pile up. 
   * This is yet again racy.  

What we can do is to increase the number of events to increase the chance of out of order event happening.